plugins/de.cau.cs.kieler.klighd.kgraph/src/de/cau/cs/kieler/klighd/kgraph/impl/EMapPropertyHolderImpl.java

```diff
@@ -174,7 +174,7 @@ public void makePersistent() {
     public <T> IPropertyHolder setProperty(final IProperty<? super T> property, final T value) {
         if (value == null) {
             getProperties().removeKey(property);
-        } else {
+        } else if (getProperties().get(property) != value) {
```

Comment:
IMO `|| property.getDefaultValue() != null` would be better here.

Reply:
I just realized that you're referring to the properties map directly instead of going via `getProperty(...)`, so default values won't be an issue, right?